hydromt/components/vector.py

```diff
@@ -232,11 +249,11 @@ def write_vector(
             Additional keyword arguments that are passed to the `write_nc`
             function.
         """
-        ds = self.vector
+        ds = self.data
         if len(ds) == 0:
-            self.logger.debug("No vector data found, skip writing.")
+            self._logger.debug("No vector data found, skip writing.")
             return
-        self._assert_write_mode()
+        self._root._assert_write_mode()
 
         # If fn is None check if vector contains only 1D data
         if fn is None:
```

Comment:
@hboisgon I was looking through this function and I figure that `fn` will never ever be `None`, not even in the old version of this function. I am wondering if there is a bunch of dead code here.

Reply:
Fixed by making fn optional